### PR TITLE
HTBHF-2498 Added v2 endpoint for dwp.

### PR DESCRIPTION
**Pull request by dwybourn**

### CODE REVIEW

File: src/main/java/uk/gov/dhsc/htbhf/productionstub/model/v2/DWPEligibilityRequestV2.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import lombok.Data;
+
+import java.time.LocalDate;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+
+@Data
+@Builder
+@AllArgsConstructor(onConstructor_ = {@JsonCreator})
+public class DWPEligibilityRequestV2 {
```
Comment (YetAnotherMatt) [inline src/main/java/uk/gov/dhsc/htbhf/productionstub/model/v2/DWPEligibilityRequestV2.java]: Couldn't you import the new DWP module to bring these classes in?

**Review by mikeSmith08 — CHANGES_REQUESTED** [review]
Yes, as Matt says in his comment, you can use the new module I've created for the request and response classes in common-java which will remove a fair amount of duplication in this PR. It will also mean you won't need to write the missing unit tests on this PR too. :-)

File: src/main/java/uk/gov/dhsc/htbhf/productionstub/controller/v2/DWPBenefitControllerV2.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        IdentityAndEligibilityResponse response = IdentityAndEligibilityResponse.builder()
+                .addressLine1Match(VerificationOutcome.NOT_MATCHED)
+                .deathVerificationFlag(DeathVerificationFlag.N_A)
+                .eligibilityStatus(EligibilityOutcome.NOT_CONFIRMED)
+                .emailAddressMatch(VerificationOutcome.NOT_MATCHED)
+                .mobilePhoneMatch(VerificationOutcome.NOT_MATCHED)
+                .postcodeMatch(VerificationOutcome.NOT_MATCHED)
+                .pregnantChildDOBMatch(VerificationOutcome.NOT_MATCHED)
+                .qualifyingBenefits(NOT_SET)
+                .identityStatus(IdentityOutcome.NOT_MATCHED)
```
Comment (mikeSmith08) [inline src/main/java/uk/gov/dhsc/htbhf/productionstub/controller/v2/DWPBenefitControllerV2.java:23]: This response isn't correct if the IdentityStatus is NOT_MATCHED, none of the other statuses should be set as they don't get checked if the identity match fails, so the response should be:
```
        return IdentityAndEligibilityResponse.builder()
                .identityStatus(IdentityOutcome.NOT_MATCHED)
                .eligibilityStatus(EligibilityOutcome.NOT_SET)
                .qualifyingBenefits(QualifyingBenefits.NOT_SET)
                .mobilePhoneMatch(VerificationOutcome.NOT_SET)
                .emailAddressMatch(VerificationOutcome.NOT_SET)
                .addressLine1Match(VerificationOutcome.NOT_SET)
                .postcodeMatch(VerificationOutcome.NOT_SET)
                .pregnantChildDOBMatch(VerificationOutcome.NOT_SET)
                .householdIdentifier(NO_HOUSEHOLD_IDENTIFIER_PROVIDED)
                .dobOfChildrenUnder4(emptyList())
                .deathVerificationFlag(DeathVerificationFlag.N_A)
                .build();
```
It is also important that the household identifier is a blank string, which is why I've used the constant `NO_HOUSEHOLD_IDENTIFIER_PROVIDED` in the code above.

Comment (dwybourn) [reply src/main/java/uk/gov/dhsc/htbhf/productionstub/controller/v2/DWPBenefitControllerV2.java:23]: Updated the response.

File: src/main/java/uk/gov/dhsc/htbhf/productionstub/model/v2/DWPEligibilityRequestV2.java (hunk 1 above)
Comment (dwybourn) [reply src/main/java/uk/gov/dhsc/htbhf/productionstub/model/v2/DWPEligibilityRequestV2.java]: Done.

File: src/test/java/uk/gov/dhsc/htbhf/productionstub/controller/v2/DWPBenefitControllerIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        assertThat(benefit.getBody().getIdentityStatus()).isEqualTo(IdentityOutcome.NOT_MATCHED);
+        assertThat(benefit.getBody().getEligibilityStatus()).isEqualTo(EligibilityOutcome.NOT_SET);
+        assertThat(benefit.getBody().getQualifyingBenefits()).isEqualTo(QualifyingBenefits.NOT_SET);
+        assertThat(benefit.getBody().getMobilePhoneMatch()).isEqualTo(VerificationOutcome.NOT_SET);
+        assertThat(benefit.getBody().getEmailAddressMatch()).isEqualTo(VerificationOutcome.NOT_SET);
+        assertThat(benefit.getBody().getAddressLine1Match()).isEqualTo(VerificationOutcome.NOT_SET);
+        assertThat(benefit.getBody().getPostcodeMatch()).isEqualTo(VerificationOutcome.NOT_SET);
+        assertThat(benefit.getBody().getPregnantChildDOBMatch()).isEqualTo(VerificationOutcome.NOT_SET);
+        assertThat(benefit.getBody().getHouseholdIdentifier()).isEqualTo("");
+        assertThat(benefit.getBody().getDobOfChildrenUnder4()).hasSize(0);
```
Comment (mikeSmith08) [inline src/test/java/uk/gov/dhsc/htbhf/productionstub/controller/v2/DWPBenefitControllerIntegrationTest.java]: You can use `isEmpty()` here.

Comment (dwybourn) [reply src/test/java/uk/gov/dhsc/htbhf/productionstub/controller/v2/DWPBenefitControllerIntegrationTest.java]: That would all null.

Comment (dwybourn) [reply src/test/java/uk/gov/dhsc/htbhf/productionstub/controller/v2/DWPBenefitControllerIntegrationTest.java]: Updated to use isEmpty().

**Review by mikeSmith08 — APPROVED** [review]

**Pull request merged**